packages/slate-react/src/components/editor.js

```diff
@@ -228,15 +176,51 @@ class Editor extends React.Component {
    */
 
   get schema() {
-    return this.state.schema
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.resolveSchema(plugins)
   }
 
   get stack() {
-    return this.state.stack
+    const plugins = this.resolvePlugins(this.props.plugins, this.props.schema)
+    return this.resolveStack(plugins)
   }
 
   get value() {
-    return this.state.value
+    if (this.tmp.value === this.props.value && this.tmp.stack === this.stack) {
+      const { value } = this.tmp
+      this.queueChange(value.change())
+      return value
+    }
+
+    return this.associateStackAndValue(this.props.value, this.stack)
+  }
+
+  /**
+   * Get value after onChange stack, it is a memoized function and therefore do not need
+   * to re-evaluate for the same value and stack.
+   *
+   * If the value is produced by onChange with the same stack, then will immediately return
+   * this value.
+   *
+   * @param {Value} value
+   * @param {Stack} stack
+   * @return {Change}
+   */
+
+  associateStackAndValue = memoizeOne((value, stack) => {
+    const change = value.change()
+    stack.run('onChange', change, this)
+    this.queueChange(change)
+    return change.value
+  })
+
+  /**
+   * Queue a change object with value, to save operations of an value
+   * @param {Change} change
+   */
+
+  queueChange(change) {
```

Comment:
Can you remove this function and inline the logic instead since it's only used in one place now?

Reply:
Fixed~